added --skip-dependencies and --only-dependencies options

ComaVN opened the pull request:
added --skip-dependencies and --only-dependencies options for facilitating installs where homebrew needs to be run as a different user.

The usecase for this is when the `developer` user does not have write permissions in /usr/local (ie. homebrew must be run as an `admin` user)

in that case, `admin` should run

```
docker-osx-dev install --only-dependencies
```

and the `developer` runs

```
docker-osx-dev install --skip-dependencies
```

This fixes issue #152

ComaVN committed:
added --skip-dependencies and --only-dependencies options for facilitating installs where homebrew needs to be run as a different user
github brikis98/docker-osx-dev issue #152

brikis98 commented on src/docker-osx-dev:
```diff
@@ -1208,6 +1236,12 @@ function handle_command {
         shift
         ;;
+      --only-dependencies)
+        ONLY_DEPENDENCIES=true
```
Instead of making these global, mutable variables, could you pass them as parameters to the `install` function?

brikis98 commented on src/docker-osx-dev:
```diff
@@ -102,6 +102,28 @@ function assert_non_empty {
+#
+# Asserts that at most one of $VAR_NAME1, $VAR_NAME2, ..., $VAR_NAMEn is defined
+#
+function assert_mutually_exclusive {
```
This is nice, but I'm not sure how many other use cases we'll have like it to merit supporting this much code. It seems like a simple `if $only_dependencies && $skip_dependencies; then log_error ...` would be enough.

ComaVN replied:
Haha, I hardly ever do anything in bash, and got carried away. Still, it's covered by tests, and could be useful down the road, so I'd leave it in.

brikis98 commented on src/docker-osx-dev:
```diff
@@ -871,6 +893,8 @@ function instructions {
   echo -e "  -e, --exclude-path PATH\t\tExclude PATH while syncing. Behaves identically to rsync's --exclude parameter. May be specified multiple times. Default: $DEFAULT_EXCLUDES"
   echo -e "  -c, --compose-file COMPOSE_FILE\tRead in this docker-compose file and sync any volumes specified in it. Default: $DEFAULT_COMPOSE_FILE"
   echo -e "  -i, --ignore-file IGNORE_FILE\t\tRead in this ignore file and exclude any paths within it while syncing (see --exclude). Default: $DEFAULT_IGNORE_FILE"
+  echo -e "  --only-dependencies\t\t\tDuring install, only install the homebrew dependencies. Useful if homebrew is needs to be run as a different user."
```
Probably worth adding a short version for consistency.

ComaVN replied:
Sure, if you want, but there's only 26 of them, and I figured this usecase was obscure enough not to spend a short option on it.

brikis98 commented:
Thanks for the PR! I added a few comments.

ComaVN committed:
Update README.md

added documentation about new --skip-dependencies and --only-dependencies flags

ComaVN committed:
improved PR #154 (fix for issue #152):
 - don't use global vars

ComaVN committed:
improved PR #154 (fix for issue #152):
 - better alignment of --help instructions

ComaVN commented:
I've improved things a bit

brikis98 commented:
Thanks!
